tests/TestCase/Controller/Component/RequestHandlerComponentTest.php

```diff
@@ -797,7 +797,7 @@ public function testCheckNotModifiedByEtagStar(): void
 
         $event = new Event('Controller.beforeRender', $this->Controller);
         $requestHandler = new RequestHandlerComponent($this->Controller->components());
-        $this->assertNull($requestHandler->beforeRender($event));
```

Comment:
Removing assertions might not be a good idea.

Reply:
They are not removed :)

Reply:
> They are nonsense from the start for void.

But `beforeRender()` callbacks can return a response instance.

Reply:
I see we have added return typehint `void` for `RequestHander::beforeRender()`. That was probably not the best idea.

Reply:
;) yeah, if we fix that, we can also undo this part here.